### DISCUSSION TITLE
How to return a warning flag variable when the simulation becomes unstable

### DISCUSSION THREAD

**Discussion by yinfanyi** [google-deepmind/mujoco#2156] [Asking for Help]
### Intro

Hi!

I am a undergrad student, I use MuJoCo for my research on Parallel mechanism simulation.

### My setup

MuJoCo version 3.2.2

### My question

My simulation sometimes encounters crashes. How can I return a variable to indicate that the simulation is unstable when a crash occurs, and then terminate the simulation instead of restarting it.

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment]
We have a WIP change which will make `mj_step` and related functions return an enum rather than void, and you could check that enum to check if anything bad happened. Would that solve your issue?

In the meantime, you can check `mjData.warning` after every step. Even though by default divergence triggers a reset, the warning array will be incremented.

**Reply by yinfanyi** [reply]
thanks a lot, it solves my issue!
